js/on-demand-scripts/qg-modal.js

```diff
@@ -0,0 +1,131 @@
+      $('a.lightbox[href^="#"]').each(function (e) {
+        $(this).addClass('content-modal');
+      }); // Detect images modals
+  
+      $('a.lightbox[href$=".png"],a.lightbox[href$=".jpg"],a.lightbox[href$=".gif"]').each(function (e) {
+        $(this).addClass('image-modal');
+      }); // Use bootstrap modals for anything with the 'lightbox' class
+  
+      $('a.lightbox').each(function (e) {
+        // Handle ajax modals
+        if (!$(this).hasClass('image-modal') && !$(this).hasClass('content-modal')) {
+          $(this).addClass('ajax-modal');
+          var ajaxUrl = $(this).attr('href');
+  
+          if (ajaxUrl.indexOf('?') !== -1) {
+            ajaxUrl += '&SQ_ASSET_CONTENTS&SQ_PAINT_LAYOUT_NAME=modal';
+          } else {
+            ajaxUrl += '?SQ_ASSET_CONTENTS&SQ_PAINT_LAYOUT_NAME=modal';
+          }
+  
+          $(this).attr('href', ajaxUrl);
+        }
+  
+        this.addEventListener('click', function (e) {
+          e.preventDefault();
+        }); // Initiate modal
+  
+        lightboxToModal($(this));
+      }); // On modal open
+  
+      $('#gov-modal').on('show.bs.modal', function (event) {
+        var button = $(event.relatedTarget);
+        var modal = $(this);
+        var url = button.data('href');
+        var title;
+        var close = '<button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>'; // Get modal title for in-page modal
+  
+        function modalTitle(content) {
+          modal.find('.modal-content').html('<div class="modal-body">' + content + '</div>');
+  
+          if (title = modal.find('.modal-body > h2:first-child').eq(0)) {
+            title.remove();
+            var titleText = title.text();
+  
+            if (title.hasClass('toc-header-number')) {
+              titleText = titleText.replace(/\d+\.\s+/, '').trim();
+            }
+  
+            modal.find('.modal-body').before('<div class="modal-header"><h2 class="modal-title mt-0" id="gov-modal--title">' + titleText + '</h2>' + close + '</div>');
+          }
+        }
+  
+        if (!button.hasClass('image-modal')) {
+          modal.removeClass('modal-img').remove('.close');
+        } // Load content from another element on the same page.
+  
+  
+        if (url.substring(0, 1) === '#') {
+          if ($(url).length) {
+            modalTitle($(url).html());
+          }
+  
+          return;
+        } // Load target image.
+  
+  
+        if (button.hasClass('image-modal')) {
+          modal.addClass('modal-img');
+          modal.find('.modal-content').html('<div class="modal-body">').prepend('<div class="modal-header">' + close + '</div>');
+          modal.find('.modal-body').html('<img src="' + url + '">');
+          return;
+        } // Load via ajax (from another page).
+  
+  
+        $.ajax({
+          url: button.data('href'),
+          // Load the content in to the modal.
+          success: function success(data) {
```

Comment:
is there any CORS error if data is loaded from different domain?

Reply:
This feature will be available on the same domain (qld.gov.au)